### MerkleSokrati_assignment2
### SKUs analyses

In [1]:
# import necessary python modules

import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
from scipy import stats

In [2]:
# read the excel dataset provided, parsing to Datetime format and formatting of unset values 

data = pd.read_excel('Data Analyst Assignment.xlsx', index_col=0, header=0, parse_dates=True, sheet_name="Assignment-2")
data.head()

2019-08-01  2019-08-02  2019-08-03  2019-08-04  2019-08-05  2019-08-06  \
SKU                                                                            
2527        1551        1613        1613        1532        1517        1441   
3042        2240        2330        2423        2350        2374        2398   
3086         891         909         936         889         880         906   
3155        1628        1628        1563        1532        1563        1563   
3166         369         354         343         340         330         337   

      2019-08-07  2019-08-08  2019-08-09  2019-08-10  ...  2020-07-22  \
SKU                                                   ...               
2527        1383        1424        1381        1340  ...        1929   
3042        2278        2210        2232        2277  ...        7096   
3086         870         879         844         852  ...         829   
3155        1532        1593        1513        1498  ...        1719   
3166         350         343         340         340  ...         335   

      2020-07-23  2020-07-24  2020-07-25  2020-07-26  2020-07-27  2020-07-28  \
SKU                                                                            
2527        1948        1929        1987        2027        1926        1926   
3042        7025        6955        6746        6881        6606        6804   
3086         829         812         788         788         772         811   
3155        1805        1895        1895        1990        2030        2050   
3166         328         325         319         316         319         329   

      2020-07-29  2020-07-30  2020-07-31  
SKU                                       
2527        1849        1904        1828  
3042        6872        6872        7078  
3086         803         763         763  
3155        2132        2111        2027  
3166         336         349         352  

[5 rows x 366 columns]

In [3]:
data = data.T

# Answer question 1
# What is the monthly & quaterly sales data by SKU?


In [4]:
pd_month = data.resample('M').sum()
pd_month = pd_month.to_period('M')
pd_quart = data.resample('3M').sum()
pd_quart = pd_quart.to_period('3M')

# Monthly sales data

In [5]:
pd_month

SKU       2527    3042   3086   3155   3166   8272   9311  13159  13160  \
2019-08  41702   69451  27412  46443  10720  31559  19045   1822   1485   
2019-09  36880   62415  31687  41922  10306  35877  17797   1990   1753   
2019-10  30582   71082  33122  45421  11966  35109  14797   1857   2045   
2019-11  21916   83246  28937  55371  13094  35407  11495   1853   1445   
2019-12  23804   87899  33444  72253  13848  22536   9002   2051   1272   
2020-01  21853   79546  35670  75218  13990  16265   8532   2297   1248   
2020-02  23170   78095  32029  61235  10517  14922   7193   2001   1154   
2020-03  30956   92606  30827  72241   9587  18244   8971   1967    949   
2020-04  38202  120947  25062  67543  12110  24042   9115   1800    845   
2020-05  44347  135342  22231  57518   9838  33069  10741   1447   1062   
2020-06  49341  157482  24184  53480   9076  22895   8341   1087   1261   
2020-07  55826  196062  26153  54253  10182  20594   6834   1150   1168   

SKU      14011  ...  PPAM05126  PPAM05156  PPAM06156  PPAM07156  PPAM08156  \
2019-08    186  ...        822        925        478       2395       3151   
2019-09    180  ...        814       1211        580       2266       3021   
2019-10    186  ...        971       1725        621       2490       3451   
2019-11    180  ...       1180       1649        573       2769       3412   
2019-12    186  ...       1168       1630        541       3346       2812   
2020-01    186  ...       1461       1339        388       3281       3499   
2020-02    174  ...       1136       1374        353       2601       4040   
2020-03    186  ...        887       1928        330       3107       5874   
2020-04    180  ...        538       1623        325       2990       7265   
2020-05    186  ...        358       1332        409       3837       8075   
2020-06    180  ...        307        943        363       3708       6403   
2020-07    186  ...        317        975        326       2566       5023   

SKU      PPAM11156  PSTM01257  PSTM01457  WIM5011B  WIM51234  
2019-08         62        682        323      2886        62  
2019-09         60        772        305      2447        60  
2019-10         62        779        357      1954        62  
2019-11         60        715        308      2035        60  
2019-12         62        715        319      1855        62  
2020-01         62        742        311      1567        62  
2020-02         58        803        295      1430        58  
2020-03         62        725        320      1458        62  
2020-04         60        704        349       984        60  
2020-05         62        797        390       904        62  
2020-06         60        796        342       766        60  
2020-07         62        698        444       816        62  

[12 rows x 247 columns]

# Quaterly sales data

In [6]:
pd_quart

SKU        2527    3042    3086    3155   3166    8272   9311  13159  13160  \
2019-08   41702   69451   27412   46443  10720   31559  19045   1822   1485   
2019-11   89378  216743   93746  142714  35366  106393  44089   5700   5243   
2020-02   68827  245540  101143  208706  38355   53723  24727   6349   3674   
2020-05  113505  348895   78120  197302  31535   75355  28827   5214   2856   
2020-08  105167  353544   50337  107733  19258   43489  15175   2237   2429   

SKU      14011  ...  PPAM05126  PPAM05156  PPAM06156  PPAM07156  PPAM08156  \
2019-08    186  ...        822        925        478       2395       3151   
2019-11    546  ...       2965       4585       1774       7525       9884   
2020-02    546  ...       3765       4343       1282       9228      10351   
2020-05    552  ...       1783       4883       1064       9934      21214   
2020-08    366  ...        624       1918        689       6274      11426   

SKU      PPAM11156  PSTM01257  PSTM01457  WIM5011B  WIM51234  
2019-08         62        682        323      2886        62  
2019-11        182       2266        970      6436       182  
2020-02        182       2260        925      4852       182  
2020-05        184       2226       1059      3346       184  
2020-08        122       1494        786      1582       122  

[5 rows x 247 columns]

# Answer question 2
# Which are the top 3 months in terms of overall sales?

In [7]:
pd_month.sum(axis=1).max()

20234570

In [8]:
pd_month.sum(axis=1).idxmax() 

Period('2020-03', 'M')

In [9]:
pd_month.sum(axis=1).sort_values(ascending=False)[0:3]

2020-03    20234570
2019-12    19536270
2020-01    19170099
Freq: M, dtype: int64

# Best 3 months on overall sales are:
 - december 2019 with 19536270
 - january 2020 with 19170099
 - march 2020 with 20234570

# Answer question 3
# Which SKUs are outliers in terms sales in any month?

In [10]:
# Statistical verification with z-score to verify if there are outliers

z = np.abs(stats.zscore(pd_month))
z = z > 3
print(np.count_nonzero(z))

0


### It seems there aren't SKU outliers on a monthly basis